src/core/ext/filters/client_channel/lb_policy.h

```diff
@@ -419,6 +434,33 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
   std::unique_ptr<ChannelControlHelper> channel_control_helper_;
 };
 
+//
+// Implementation
+//
+
+template <typename T>
```

Comment:
I copy/pasted some fragments from the promises CL to make this:
https://godbolt.org/z/sWK3n7zcn

Suggest we move in the Overload type, and probably the associated Match function as a way of generalizing this part of the CL.

I can do so tomorrow if you'd like.

Reply:
Sure, if you want to throw together a PR to add this `Match()` function somewhere in src/core/lib/gprpp, we can use that here.

Reply:
Ping?  I'm happy to change this PR to use that sort of `Match()` function if you want to send me a PR to create it.

Reply:
#26640

Reply:
(sorry about the delay, detangling stuff from the promises PR got tricky)

Reply:
I've tried changing this to use the new `Match()` API, but the compiler is generating lock annotation errors.  It looks like something in the templating is breaking the lock annotation flow here.  Any idea how to fix this?

Reply:
As we discussed, I changed this back to using a custom helper function that doesn't break the lock annotation, but I've kept it out of the LB policy API, and I've added a TODO about changing it to use `Match()` once we find a way to not break the lock annotations.